app/screens/modal/NoInternetConnection.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Button } from '../../basicComponents';
+import { ExternalLinks } from '../../../shared/constants';
+import Modal from '../../components/common/Modal';
+import ReactPortal from './ReactPortal';
+
+const ButtonsWrapper = styled.div<{ hasSingleButton?: boolean }>`
+  display: flex;
+  flex-direction: row;
+  justify-content: ${({ hasSingleButton }) =>
+    hasSingleButton ? 'center' : 'space-between'};
+  margin: auto 0 15px 0;
+  padding-top: 30px;
+`;
+
+const ErrorMessage = styled.pre`
+  font-size: 14px;
+  line-height: 16px;
+  word-wrap: break-word;
+  white-space: pre-wrap;
+  flex: 1;
+  overflow-y: auto;
+`;
+const getOnLineStatus = () =>
+  typeof navigator !== 'undefined' && typeof navigator.onLine === 'boolean'
+    ? navigator.onLine
+    : true;
+
+const useNavigatorOnLine = () => {
+  const [status, setStatus] = React.useState(getOnLineStatus());
+
+  React.useEffect(() => {
+    const setOnline = () => setStatus(true);
+    const setOffline = () => setStatus(false);
+    window.removeEventListener('online', setOnline);
+    window.removeEventListener('offline', setOffline);
+
+    window.addEventListener('online', setOnline);
+    window.addEventListener('offline', setOffline);
```

Comment:
Instead of unsubscribing from events first it will be much better to return the unsubscribe function there. Then Component will unsubscribe from these events on unmounting...

```suggestion
    const setOnline = () => setStatus(true);
    const setOffline = () => setStatus(false);
    window.addEventListener('online', setOnline);
    window.addEventListener('offline', setOffline);
    return () => {
      window.removeEventListener('online', setOnline);
      window.removeEventListener('offline', setOffline);
    };
```

Reply:
yeah, but exactly with return () => , the functionality won't be working for some reason. I think maybe it is because it is on a low-level React that has some priority to call. I will check this idea

Reply:
Btw calling `removeEventListener` in such a context is useless, because you've just created new functions (event listener has a reference to another handler, it won't remove event listeners even if it already has some).

Reply:
If it does not work when you return a function that removes event listeners — then I bet something with the mounting/unmounting component. Maybe because of ReactPortal.